src/main/java/com/epam/task/first/logic/ArrayLogic.java

```diff
@@ -0,0 +1,91 @@
+package com.epam.task.first.logic;
+
+import com.epam.task.first.entities.Array;
+
+import java.util.List;
+
+
+public class ArrayLogic {
+
+    public int findMax(final Array<Integer> a) {
+        List<Integer> elements = a.getElements();
+        if (elements.isEmpty()) {
+            throw new IllegalArgumentException();
+        }
+        return getMax(elements);
+    }
+
+    public double countAverage(final Array<Integer> a) {
+        List<Integer> elements = a.getElements();
```

Comment:
duplication, you can create method checkArray(array), but I would say don't such such checks, and dont use runtime exceptions for now